lib/src/Uncertainty/Algorithm/Simulation/openturns/PhysicalSpaceCrossEntropyImportanceSampling.hxx

```diff
@@ -0,0 +1,90 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief PhysicalSpaceCrossEntropyImportanceSampling implements Importance Sampling by Cross Entropy algorithm in Physical Space
+ *
+ *  Copyright 2005-2023 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_PHYSICALSPACECROSSENTROPYIMPORTANCESAMPLING_HXX
+#define OPENTURNS_PHYSICALSPACECROSSENTROPYIMPORTANCESAMPLING_HXX
+
+
+#include "openturns/CrossEntropyImportanceSampling.hxx"
+#include "openturns/OptimizationAlgorithm.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class PhysicalSpaceCrossEntropyImportanceSampling
+ *
+ * PhysicalSpaceCrossEntropyImportanceSampling is some PhysicalSpaceCrossEntropyImportanceSampling type to illustrate how to add some classes in OpenTURNS
+ */
+
+
+class OT_API PhysicalSpaceCrossEntropyImportanceSampling
+  : public CrossEntropyImportanceSampling
+{
+CLASSNAME
+public:
+/** Default constructor */
+  PhysicalSpaceCrossEntropyImportanceSampling();
+
+  /** Default constructor */
+  PhysicalSpaceCrossEntropyImportanceSampling(const RandomVector & event,
+                                              const Distribution &auxiliaryDistribution,
+                                              const Indices &activeParameters,
+                                              const Point &initialAuxiliaryDistributionParameters,
+                                              const Interval &bounds,
+                                              const Scalar rhoQuantile = ResourceMap::GetAsScalar("CrossEntropyImportanceSampling-DefaultRhoQuantile"));
+
+  /** Virtual constructor */
+  PhysicalSpaceCrossEntropyImportanceSampling * clone() const override;
+  
+  /** Set solver */
+  void setSolver(const OptimizationAlgorithm & solver);
+  
+  
+  /** Get solver */
+  OptimizationAlgorithm getSolver() const;
```

Comment:
in many objects such a method called getOptimizationAlgorithm

Reply:
ok, thanks.